lib/zclient.c

```diff
@@ -2467,11 +2476,13 @@ struct interface *zebra_interface_link_params_read(struct stream *s,
 		return NULL;
 	}
 
-	if (if_link_params_get(ifp)) {
+	iflp = if_link_params_get(ifp);
+
+	if (iflp) {
 		iflp_prev_set = true;
-		memcpy(&iflp_prev, ifp->link_params, sizeof(iflp_prev));
-	} else
-		iflp_prev_set = false;
+		admin_group_init(&iflp_prev.ext_admin_grp);
+		if_link_params_copy(&iflp_prev, iflp);
+	}
```

Comment:
I would prefer to keep the memcpy() call and add one more call if extended admin group is set instead of calling a dedicated function which copy field by field for performance and code safe

Reply:
code changed